src/commands/lightning/preview/app.ts

```diff
@@ -202,46 +204,68 @@ export default class LightningPreviewApp extends SfCommand<void> {
       targetOrg = this.argv[idx + 1];
     }
 
+    // Start the LWC Dev Server
+    const lwcServer = await startLWCServer(sfdxProjectRootPath, logger);
+    const serverPort = lwcServer.context?.config?.port ?? LOCAL_DEV_SERVER_DEFAULT_PORT;
```

Comment:
That feels like the long way around to retrieving information that the plugin itself is providing. Should we not read this value from our configuration?

Reply:
But then a lot of the stuff around the setup of the dev server feels disconnected, in general. We've hidden a lot of "public API" details inside of function implementations that initialize not-objects.

It feels to me like:

- The dev server should be an object/type/class: say `LWCDevServer`.
- It should be able to take configuration items that exist in its public API (e.g. ports, project directories, loggers), in its constructor.
- It should use the implementation of its constructor for a lot of the bootstrapping that otherwise happens through code in `createLWCServerConfig()`.

That way you don't end up in situations where you have to query the object you just created, to get the data back the you ostensibly would have otherwise passed into it.

Sorry, probably not actionable in this PR. But that section could use some refactor.